homeassistant/components/wmspro/generic_entity.py

```diff
@@ -0,0 +1,51 @@
+"""Generic entity for the WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+from wmspro.destination import Destination
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+
+class WebControlProGenericEntity(Entity):
+    """Foundation of all WMS based entities."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, dest: Destination) -> None:
+        """Initialize the entity with destination channel."""
+        super().__init__()
+        self._dest = dest
+
+    async def async_update(self) -> None:
+        """Update the entity."""
+        await self._dest.refresh()
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self.__class__.__name__}_{self._dest.id}"
```

Comment:
Let's not include a class name in the unique_id

Reply:
I will change this, but adopted this approach from the HomematicIP integration.